client/lib/analytics/fullstory.js

```diff
@@ -10,12 +11,17 @@ import config from '@automattic/calypso-config';
 import { mayWeTrackCurrentUserGdpr, isPiiUrl } from './utils';
 import { getCurrentUser, getDoNotTrack } from '@automattic/calypso-analytics';
 import { isE2ETest } from 'calypso/lib/e2e';
+import { TRACKING_IDS } from './ad-tracking/constants';
 
 const fullStoryDebug = debug( 'calypso:analytics:fullstory' );
 
 let fullStoryScriptLoaded = false;
 
 export function retargetFullStory() {
+	if ( document.location.href.indexOf( 'checkout' ) !== -1 ) {
+		return;
```

Comment:
I think this should call `window.FS.shutdown()` before returning. My suggestion is to leave this out for now and return to it together with the consideration for VIP sites. Another thing we'll need to look carefully at is how this will impact events. If we do an early return here upon catching the `checkout` exception, it prevents the retargeting pageview event, but that doesn't prevent `recordFullStoryEvent()` from being fired in other contexts.